### DISCUSSION TITLE
Classification of Interaction Points / Ports?

### DISCUSSION THREAD

**Discussion by parkaneric** [GfSE/SAF-Specification#92] [Ideas]
As SAF is for architectures of (complex) cyber-mechatronical systems I think it is worth to consider a concept to classify interaction points i.e. ports in the functional/logical as well as the phyiscal domain.
E.g. VDA / AIAG Handbook propagates different kinds of interfaces or interaction points:
- physical contact
- exchange of material
- transmission of energy
- exchange of information
- man-machine-interface
- adjacent systems
This enables also a color coding for ibds, which improves the diagram readability.
What do you think?

**Comment by haarer** [comment]
Am 18. Dezember 2024 08:18:45 MEZ schrieb Eric Parkan ***@***.***>:
>As SAF is for architectures of (complex) cyber-mechatronical systems I think it is worth to consider a concept to classify interaction points i.e. ports in the functional/logical as well as the phyiscal domain.
>E.g. VDA / AIAG Handbook propagates different kinds of interfaces or interaction points:
>- physical contact
>- exchange of material
>- transmission of energy
>- exchange of information
>- man-machine-interface
>- adjacent systems
>This enables also a color coding for ibds, which improves the diagram readability.
>What do you think?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/GfSE/SAF-Specification/discussions/92
>You are receiving this because you are subscribed to this thread.
>
>Message ID: ***@***.***>
I like the idea of assigning some Kind of classification to interaction Points.

I believe we shouldn't force SAF Users into a specific classification scheme, but provide  a mechanism, which is extensible. We could provide some typical classifications in a library.

Classifications could IMO also be multi dimensional, e.g. have an engineering discipline and an other dimension, e.g. for the ownership of the interface definition.

The idea could also be extended to the types used in interaction points, possibly allowing a rollup of type classifications into the interaction point. 

What do you think?
What are your use cases?




-- 
Gruß,
Alexander

**Reply by parkaneric** [reply]
Good point. However, in the moment SAF does not contain any library mechanism to extend the profile. So this is definitivly new and should considered very carefully.

**Reply by parkaneric** [reply]
Usecase-wise I think it really depends on the domain:
1. for functional/logical (i.e. coneptual) I want to highlight a need for an dedicated interface which I can use to trace requirements to. Thus, the classification of the interaction point helps to specify (and communicate) the need.
2. for physical I want to document the implementation of interfaces. Following P5_PIFD and P4_PIEX I would use the interaction point classification to improve this documentation when multiple aspects of the interaction points are presented.

**Comment by haarer** [comment]
Am 19. Dezember 2024 08:31:16 MEZ schrieb Eric Parkan ***@***.***>:
>Good point. However, in the moment SAF does not contain any library mechanism to extend the profile. So this is definitivly new and should considered very carefully.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/GfSE/SAF-Specification/discussions/92#discussioncomment-11613984
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
I dont want the profile itself to be extended.

The addon should not contain stereotypes, but items stereotyped by some to be defined 'classification' stereotype. E.g. the VDA definitions you mentioned.

Someone could contribute a library using SAF standards VP and e.g.  put VDA terms in it along with VDA defined interface classifications.

We could add such libraries as 'contributed' addon to the profile repository.

The library used in the cameo example  works in a similar way, it is installed with the cameo plugin.
-- 
Gruß,
Alexander
